instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/datasource/OpenTelemetryDataSource.java

```diff
@@ -66,12 +66,29 @@ public OpenTelemetryDataSource(DataSource delegate) {
    * @param delegate the DataSource to wrap
    * @param openTelemetry the OpenTelemetry instance to setup for
    */
+  @Deprecated
   public OpenTelemetryDataSource(DataSource delegate, OpenTelemetry openTelemetry) {
     this.delegate = delegate;
-    this.dataSourceInstrumenter = createDataSourceInstrumenter(openTelemetry);
+    this.dataSourceInstrumenter = createDataSourceInstrumenter(openTelemetry, true);
     this.statementInstrumenter = createStatementInstrumenter(openTelemetry);
   }
 
+  /**
+   * Create a OpenTelemetry DataSource wrapping another DataSource.
+   *
+   * @param delegate the DataSource to wrap
+   * @param dataSourceInstrumenter the DataSource Instrumenter to use
+   * @param statementInstrumenter the Statement Instrumenter to use
+   */
+  public OpenTelemetryDataSource(
```

Comment:
could be package private

Reply:
This would be a break change though for some people. Are we sure we want to do this right now?

Reply:
this is the constructor you added

Reply:
My mistake. I'll update this soon.